app/furniture/marketplace/delivery_area.rb

```diff
@@ -22,7 +22,15 @@ def discardable?
     end
 
     def destroyable?
-      persisted? && discarded? && orders.empty?
+      persisted? && discarded? && unordered?
```

Comment:
What about adding a new `.safe_destroy` method to the model and putting the check therein, so you wouldn't have to write `delivery_area.destroy if delivery_area.destroyable?` ? I'm not worried about the extra keystrokes so much as making it easier to avoid a mistake. Does creating a method like that contradict conventional ActiveRecord model configuration?

Reply:
I could see something like `destroy_safely` which:
- Discards if it's `discardable?`
- Destroys if it's `destroyable?`
- Does nothing if either.

Reply:
Maybe. I worry a bit about losing the explicit intent to discard vs destroy.

Reply:
You're right, it's better for `destroy_safely`  to be `destroy if destroyable?` and otherwise do nothing.